lua/orgmode/org/links/init.lua

```diff
@@ -0,0 +1,130 @@
+local utils = require('orgmode.utils')
+local Link = require('orgmode.org.links.link_handler')
+
+---@class OrgLinkHandlerRegistry
+---@field private internal OrgLinkHandler
+---@field private handlers OrgLinkHandler[]
+---@field private handlers_by_name table<string, OrgLinkHandler>
+---@field private stored_links OrgLinkHandler[]
+local OrgLinkHandlerRegistry = {}
+OrgLinkHandlerRegistry.__index = OrgLinkHandlerRegistry
+
+-- Using `linkword` definition as defined in https://orgmode.org/org.pdf#Link%20Abbreviations
+local protocol_deliniator_pattern = '^[a-z][a-z0-9-_]+:'
+
+function OrgLinkHandlerRegistry:new(opts)
+  local this = setmetatable({
+    internal = nil,
+    handlers = {},
+    handlers_by_name = {},
+    stored_links = {},
+  }, self)
+  self.__index = self
+  this:setup_builtin_handlers()
+end
+
+function OrgLinkHandlerRegistry:setup_builtin_handlers()
+  self:add_handler(require('orgmode.org.links.handlers.file'))
+  self:add_handler(require('orgmode.org.links.handlers.http'))
+  self:add_handler(require('orgmode.org.links.handlers.https'))
+  self:add_handler(require('orgmode.org.links.handlers.id'))
+  self.internal = require('orgmode.org.links.handlers.internal')
+end
+
+---@param handler OrgLinkHandler
+function OrgLinkHandlerRegistry:add_handler(handler)
+  if self.handlers_by_name[handler.protocol] then
+    error('Completion handler ' .. handler.protocol .. ' already exists')
+  end
+
+  -- Ensure that all handlers have defaults for the methods we rely on.
+  for k, v in pairs(Link) do
+    handler[k] = handler[k] or v
+  end
```

Comment:
We don't need this for now since we are not exposing adding custom handlers yet.

Reply:
It's just as valid for protecting against internal handler's issues as it is for protecting custom handlers